versions/1.1/SPEC.md

```diff
@@ -0,0 +1,5846 @@
+  }
+  runtime {
+    container: "my_image:latest"
+    memory: "~{memory_mb + 256} MB"
+  }
+}
+```
+
+#### Example 4: BWA MEM
+
+```wdl
+task bwa_mem_tool {
+  input {
+    Int threads
+    Int min_seed_length
+    Array[Int]+ min_std_max_min
+    File reference
+    File reads
+  }
+  command {
+    bwa mem -t ${threads} \
+            -k ${min_seed_length} \
+            -I ${sep=',' min_std_max_min} \
+            ${reference} \
+            ${sep=' ' reads+} > output.sam
+  }
+  output {
+    File sam = "output.sam"
+  }
+  runtime {
+    container: "my_image:latest"
+    cpu: threads
+  }
+}
+```
+
+#### Example 5: Word Count
+
+```wdl
+task wc2_tool {
+  input {
+    File file1
+  }
+  command {
+    wc ${file1}
+  }
+  output {
+    Int count = read_int(stdout())
+  }
+  runtime {
+    container: "my_image:latest"
+  }
+}
+
+workflow count_lines4_wf {
+  input {
+    Array[File] files
+  }
+  scatter (f in files) {
+    call wc2_tool {
+      input: file1 = f
+    }
+  }
+  output {
+    wc2_tool.count
+  }
+}
+```
+
+#### Example 6: tmap
+
+```wdl
+task tmap_tool {
+  input {
+    Array[String] stages
+    File reads
+  }
+  command {
+    tmap mapall ${sep=' ' stages} < ${reads} > output.sam
+  }
+  output {
+    File sam = "output.sam"
+  }
+  runtime {
+    container: "my_image:latest"
+  }
+}
+```
+
+Given the following inputs:
+
+|Variable|Value|
+|--------|-----|
+|reads   |/path/to/fastq|
+|stages  |["stage1 map1 --min-seq-length 20 map2 --min-seq-length 20", "stage2 map1 --max-seq-length 20 --min-seq-length 10 --seed-length 16  map2 --max-seed-hits -1 --max-seq-length 20 --min-seq-length 10"]|
+
+This task produces a command line like this:
+
+```sh
+tmap mapall \
+stage1 map1 --min-seq-length 20 \
+       map2 --min-seq-length 20 \
+stage2 map1 --max-seq-length 20 --min-seq-length 10 --seed-length 16 \
+       map2 --max-seed-hits -1 --max-seq-length 20 --min-seq-length 10
+```
+
+## Workflow Definition
+
+
+```wdl
+workflow name {
+  input {
+    # workflow inputs are declared here
+  }
+
+  # other "private" declarations can be made here
+ 
+  # there may be any number of (potentially nested) 
+  # calls, scatter blocks, or conditional blocks
+  call target { input: ... }
+  scatter (i in collection) { ... }
+  if (condition) { ... }
+
+  output {
+    # workflow outputs are declared here
+  }
+
+  meta {
+    # workflow-level metadata can go here
+  }
+
+  parameter_meta {
+    # metadata about each input/output parameter can go here
+  }
+}
+```
+
+A workflow can be thought of as a directed acyclic graph (DAG) of transformations that convert the input data to the desired outputs. Rather than explicitly specifying the sequence of operations, A WDL workflow instead describes the connections between the steps in the workflow (i.e. between the nodes in the graph). It is the responsibility of the execution engine to determine the proper ordering of the workflow steps, and to orchestrate the execution of the different steps.
```

Comment:
I'd put this above the code chunk above.  You'd just scroll through the chunk, then read the text, then scroll back up.

Reply:
done